crates/swc_ecma_transforms_typescript/tests/fixture/issue-7104/case1/output.stderr

```diff
@@ -0,0 +1,14 @@
+
+  x An arrow function is not allowed here
+   ,-[input.js:1:1]
+ 1 | const foo = <T extends {}>() => {
```

Comment:
I didn't understand why this error come out,
I expect we get the same error message as `case2`.